redfish-core/lib/assembly.hpp

```diff
@@ -325,6 +334,103 @@ inline void handleChassisAssemblyGet(
     });
 }
 
+/**
+ * @brief Set location indicator for the assemblies associated to given chassis
+ * @param[in] req - The request data
+ * @param[in] asyncResp - Shared pointer for asynchronous calls.
+ * @param[in] chassis - Chassis the assemblies are associated with.
+ * @param[in] assemblies - list of all the assemblies associated with the
+ * chassis.
+
+ * @return None.
+ */
+inline void setAssemblylocationIndicators(
+    const crow::Request& req,
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& chassisID, const std::vector<std::string>& assemblies)
+{
+    BMCWEB_LOG_DEBUG(
+        "Set locationIndicator for assembly associated to chassis = {}",
+        chassisID);
+
+    std::optional<std::vector<nlohmann::json>> assemblyData;
+    if (!json_util::readJsonAction(req, asyncResp->res, "Assemblies",
+                                   assemblyData))
+    {
+        return;
+    }
+    if (!assemblyData)
+    {
+        return;
+    }
+
+    std::vector<nlohmann::json> items = std::move(*assemblyData);
+    std::map<std::string, bool> locationIndicatorActiveMap;
+
+    for (auto& item : items)
+    {
+        std::optional<std::string> memberId;
+        std::optional<bool> locationIndicatorActive;
+
+        if (!json_util::readJson(item, asyncResp->res,
+                                 "LocationIndicatorActive",
+                                 locationIndicatorActive, "MemberId", memberId))
+        {
+            return;
+        }
+        if (locationIndicatorActive)
```

Comment:
Yes, it is functionally equivalent to 1060.

The input array may contain the values of LocationIndicatorActive, or null -e.g.
```
[ {"MemberId" : "0", "LocationIndicatorActive" : true} , {"MemberId" : "1", "LocationIndicatorActive" : true} ,  ...]
```
This particular lines are reading one of the element, and check whether the field `LocationIndicatorActive' is specified or not, and then build `locationIndicatorActiveMap`.  

The logic is trying to change the only-specified  elements, not all of the elements  in assembly.
The map is to gather the specified elements  so that they are the only ones to change.

Reply:
I'll check more about optional<bool>... Probably I may need to change it to  `const bool *`.